lib/view_component/base.rb

```diff
@@ -296,6 +301,20 @@ def templates
           end
       end
 
+      def parameter_errors
+        @parameter_errors ||=
+          begin
+            errors = []
+
+            # If initializer omits with_collection_parameter
+            if @with_collection_parameter && !instance_method(:initialize).parameters.map(&:last).include?(@with_collection_parameter)
+              errors << "Collection parameter #{@with_collection_parameter} not used in component initializer."
```

Comment:
Nit Pik The use of the instance variable here feels a little icky to me. I wonder if we can't put an attr_reader or the like around it.